octree/include/pcl/octree/impl/octree_pointcloud.hpp

```diff
@@ -39,11 +39,10 @@
 #ifndef PCL_OCTREE_POINTCLOUD_HPP_
 #define PCL_OCTREE_POINTCLOUD_HPP_
 
-#include <vector>
 #include <assert.h>
 
 #include <pcl/common/common.h>
-
+#include <pcl/octree/impl/octree_base.hpp>
```

Comment:
This one is a bit tricky.
Actually including neither octree_base.h nor octree_base.hpp should not be necessary at this point. But since OctreePointCloudAdjacency, for example, is used somewhere else in PCL with a custom LeafContainer and PCL_NO_PRECOMPILE is turned off, OctreePointCloudAdjacency is not instantiated for these LeafContainers. That's why octree_pointcloud_adjacency.hpp includes octree_pointcloud.hpp which in turn includes the octree_base.hpp in order to provide the implementation for that case.
However, perhaps it would make more sense to change the other code somewhere in PCL (don't look up the location, yet) to always include the required octree hpp files and remove the nested hpp includes in the octree module. That would yield a uniform include structure in the octree module.
Hope my thoughts are understandable.